chia/wallet/wallet_protocol.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List, Optional, Set
+
+from blspy import G1Element
+from typing_extensions import Protocol
+
+from chia.server.ws_connection import WSChiaConnection
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.ints import uint8, uint32, uint64, uint128
+from chia.wallet.wallet_coin_record import WalletCoinRecord
+
+if TYPE_CHECKING:
+    from chia.wallet.wallet_state_manager import WalletStateManager
+
+
+class WalletProtocol(Protocol):
```

Comment:
Did you verify that all the wallet classes actually get complained about if you tweak the protocol in each of a few incompatible ways?

Reply:
I did not, and I'm not sure it matters. No wallet declares that it implements the protocol, this is just a hint to mypy. mypy did complain in a bunch of places. but mypy is still partially disabled in a lot of the wallet code, so I wouldn't expect it to catch everything yet either.

Reply:
I added checks